integration/smartbft/mock_orderer.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package smartbft
+
+import (
+	"context"
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	ab "github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/util"
+	"github.com/hyperledger/fabric/internal/pkg/comm"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+	"io"
+	"math"
+	"time"
+)
+
+type MockOrderer struct {
+	address     string
+	ledgerArray []*common.Block
+	logger      *flogging.FabricLogger
+	grpcServer  *comm.GRPCServer
+	channel     chan string
+}
+
+func (mo *MockOrderer) parseEnvelope(ctx context.Context, envelope *cb.Envelope) (*cb.Payload, *cb.ChannelHeader, *cb.SignatureHeader, error) {
+	payload, err := protoutil.UnmarshalPayload(envelope.Payload)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if payload.Header == nil {
+		return nil, nil, nil, errors.New("envelope has no header")
+	}
+
+	chdr, err := protoutil.UnmarshalChannelHeader(payload.Header.ChannelHeader)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	shdr, err := protoutil.UnmarshalSignatureHeader(payload.Header.SignatureHeader)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return payload, chdr, shdr, nil
+}
+
+func (mo *MockOrderer) Broadcast(server ab.AtomicBroadcast_BroadcastServer) error {
+	mo.logger.Infof("Broadcast called")
+	panic("implement me: Broadcast")
+	return nil
+}
+
+func (mo *MockOrderer) Deliver(server ab.AtomicBroadcast_DeliverServer) error {
+	mo.logger.Infof("Deliver called")
+	ctx := server.Context()
+	addr := util.ExtractRemoteAddress(ctx)
+	mo.logger.Infof("Starting new deliver loop for %s", addr)
+
+	/* Create DeliverServer here */
+
+	for {
+		mo.logger.Infof("Attempting to read seek info message from %s", addr)
+		envelope, err := server.Recv()
+		if err == io.EOF {
+			mo.logger.Debugf("Received EOF from %s, hangup", addr)
+			return nil
+		}
+		if err != nil {
+			mo.logger.Warningf("Error reading from %s: %s", addr, err)
+			return err
+		}
+
+		status, err := mo.deliverBlocks(ctx, server, envelope)
```

Comment:
This delivers blocks like a correct orderer. This is good for the first test case.
In the next PR we'll add byzantine behavior here.